### ISSUE TITLE
No translation of default values for translated fields

### ISSUE THREAD

**Issue by Yajo** [odoo/odoo#9188]
You can see a real case of this problem in https://github.com/OCA/website/pull/87.

tl;dr This will not write translations for "A default" when a new record is created in the UI. It will always appear in English and without translations:

```
class A(models.Model):
    name = fields.Char(translate=True, default="A default", help="A help")
```

Also this will not write translations to the field, but will always return the default value in context lang without writing any translations:

```
class A(models.Model):
    name = fields.Char(translate=True, default=lambda self: _("A default"), help="A help")
```

Funnier is that "A help" will get translated. This is fine, but makes the ORM seem somehow inconsistent.

**Comment by michaelkarrer81** [comment]
Just had the same problem today. Is there any workaround already?

**Comment by Yajo** [comment]
Not AFAIK.

**Comment by Yenthe666** [comment]
@mart-e this might be a perfect one for you.

**Comment by mart-e** [comment]
Hello,

You are right, the `default` attribute is not translated so far. The reason is that the default is specified on the field and can either be callable, either static. In your comparison, the `help` attribute is static and stored on the `ir.model.field` object.

To make the `default` automatically translated too, we would need to also store it on the `ir.model.field` object. Not impossible but not trivial due to the callable/static nature.

**Comment by Yajo** [comment]
There's still one more problem: when you develop an addon and add some translations to its `default`s, you as a developer expect that if the user creates a record and does not change a value, the user gets it in his language, but also gets it translated in the languages where there's a translation available.

Currently the default only applies to current user language, and applies that to every language in database, translating nothing else.

**Comment by mart-e** [comment]
I have discussed this and so far we decided the use case is not frequent enough to add the complexity. I searched in our codebase and found only [survey](https://github.com/odoo/odoo/blob/7427825790daff2ee6d6b1db8c7ea73c29f7f7bc/addons/survey/models/survey.py#L473) used translated fields with default value (and uses the `lambda` solution).

I disagree with you on the expectation to have translated in every language available. Translations are never easy and I want to avoid this kind of blackmagic. Remember that we are talking about default value here, it is suggestions that the user can change before saving (when no translation is generated yet).

So far, I think that the `default=lambda s: _('...')` is an acceptable and working solution. It has also the advantage of being explicit.

So, it is not a bad idea, just that we are not going to implement it at the moment.
